hermes/probe/metrics/metrics.go

```diff
@@ -0,0 +1,183 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove, GitHub: evanSpendlove.
+//
+// Metrics implements the latency distribution metrics for Hermes.
+// Package probe implements the probe that Hermes uses to monitor
+// a storage system.
+
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/cloudprober/metrics"
+
+	probepb "github.com/googleinterns/step224-2020/config/proto"
+)
+
+// ProbeOperation represents a possible probe operation metric label.
+type ProbeOperation int
+
+const (
+	// TOTAL_PROBE_RUN is the metric label for the overall probe run.
+	TOTAL_PROBE_RUN ProbeOperation = iota
+	// CHECK_NIL is the metric label for the check nil file existence and consistency operation.
+	CHECK_NIL
+	// READ_AND_VERIFY is the metric label for the read file and verify contents operation.
+	READ_AND_VERIFY
+	// DELETE_FILE is the metric label for the delete file operation.
+	DELETE_FILE
+	// CREATE_FILE is the metric label for the create file operation.
+	CREATE_FILE
+)
+
+// ApiCall represents a possible api call metric label.
+type ApiCall int
+
+const (
+	// API_LIST_FILES is the metric label for the list files API call.
+	API_LIST_FILES ApiCall = iota
+	// API_CREATE_FILE is the metric label for the create file API call.
+	API_CREATE_FILE
+	// API_DELETE_FILE is the metric label for the delete file API call.
+	API_DELETE_FILE
+	// API_GET_FILE is the metric label for the get file API call.
+	API_GET_FILE
+)
+
+// ExitStatus represents a possible exit status metric label.
+type ExitStatus int
+
+const (
+	// SUCCESS indicates the operation/API call was successful.
+	SUCCESS ExitStatus = iota
+	// OP_TIMEOUT indicates that the operation/API call timed out.
+	OP_TIMEOUT
+	// PROBE_FAILED indicates that the probe failed due to an error.
+	PROBE_FAILED
+	// API_CALL_FAILED indicates that the API call failed due to an error.
+	API_CALL_FAILED
+	// FILE_MISSING indicates that the target file could not be found.
+	FILE_MISSING
+	// BUCKET_MISSING indicates that the target bucket could not be found.
+	BUCKET_MISSING
+	// FILE_CORRUPTED indicates that the target file was corrupted and the contents could not be read.
+	FILE_CORRUPTED
+	// FILE_READ_FAILURE indicates that the target file could not be read.
+	FILE_READ_FAILURE
+	// TODO(evanSpendlove): Review if we need this? When would we use it?
+	// FILE_METADATA_MISMATCH indicates that the target file did not match the expectation from its metadata
+	FILE_METADATA_MISMATCH
+	// UNKNOWN_FILE_FOUND indicates that an unknown file, not created by Hermes, was found in the target bucket.
+	UNKNOWN_FILE_FOUND
+	// ALL_FILES_MISSING indicates that all of the Hermes files were missing.
+	ALL_FILES_MISSING
+)
+
+var (
+	// ProbeOps maps ProbeOperation constants to their metric label string equivalent.
+	ProbeOps = map[ProbeOperation]string{
```

Comment:
Optional: I would call this map ProbeOpName.
Same for the other maps.

Reply:
I've renamed these to "_Name"